lib/dry/logic/result.rb

```diff
@@ -56,6 +56,19 @@ def success?
         end
       end
 
+      class Result::EvaluatedValue < Result
```

Comment:
How about calling it `Result::Lazy`? It's not evaluated until you ask for its `input` :)